src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -998,6 +998,191 @@ public with sharing class ADDR_Addresses_TEST {
+        Contact con = listConT[0];
+        con.put('MailingStateCode', stateCode);
+        con.is_Address_Override__c = true;
+
+        Test.startTest();
+        update con;
+        Test.stopTest();
+
+        // Verify a new Address record was created
+        List<Address__c> retrievedAddresses = getAddresses();
+        System.assertEquals(2, retrievedAddresses.size());
+
+        // Find the default Address
+        Address__c defaultAddress = null;
+        Address__c newAddress = null;
+        for (Address__c eachAddress : retrievedAddresses) {
+            if (eachAddress.Default_Address__c) {
+                defaultAddress = eachAddress;
+            } else {
+                newAddress = eachAddress;
+            }
+        }
+
+        // Verify the Account's Billing Address info matches the Default Address record
+        assertEquals(listAddrT[0], defaultAddress);
+
+        // Verify the Contact's Address was updated as expected and that the Household and Contact Addresses are different.
+        List<Account> retrievedAccounts = Database.query(buildAccountSoql());
+        Map<Id, Account> accountIdToAccount = new Map<Id, Account>(retrievedAccounts);
+        List<Contact> listCon = Database.query(buildContactSoql());
+
+        System.assertEquals(state, listCon[0].MailingState);
+        System.assertEquals(true, listCon[0].is_Address_Override__c);
+        System.assertNotEquals(null, listCon[0].Current_Address__c);
+
+        Account eachContactAccount = accountIdToAccount.get(listCon[0].AccountId);
+        System.assertEquals(false, isMatchAddressAccCon(eachContactAccount, listCon[0]));
+
+        // Verify the new Address Mailing State matches Contact's MailingStateCode and not set as the Default
+        System.assertEquals(stateCode, newAddress.MailingState__c);
+        System.assertEquals(false, newAddress.Default_Address__c);
+    }
+
+
+    /*********************************************************************************************************
+    @description
+        Update Address' Mailing Country when it is empty and State & Country Picklists are enabled.
+    verify:
+        Does not populate the Address' Mailing Country if the source sObject's CountryCode is empty
+        Populates the Address' Mailing Country with the source sObject's CountryCode value if the Mailing Country is empty
+    **********************************************************************************************************/
+    static testMethod void testEmptyAddressCountryIsPopulatedFromCountryCode() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testEmptyAddressCountryIsPopulatedFromCountryCode') {
+            return;
+        }
+
+        Contact con = new Contact();
+        con.put('MailingCountryCode', null);
+        Address__c addressEmptyCountry = new Address__c(MailingCountry__c = null);
```

Comment:
@joy-sfdo , this a great example of writing faster tests by using Sobjects that do not have to be created in DB. Thanks!

Reply:
Thanks! Great suggestion!